backend/src/main/java/pl/zzpj/esportbetting/impl/BetServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package pl.zzpj.esportbetting.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.zzpj.esportbetting.enumerate.DetailedFinishedStatusEnum;
+import pl.zzpj.esportbetting.exception.ObjectNotFoundException;
+import pl.zzpj.esportbetting.interfaces.BetService;
+import pl.zzpj.esportbetting.interfaces.UserLevelService;
+import pl.zzpj.esportbetting.model.Bet;
+import pl.zzpj.esportbetting.model.Match;
+import pl.zzpj.esportbetting.model.User;
+import pl.zzpj.esportbetting.repos.BetRepository;
+import pl.zzpj.esportbetting.repos.UserRepository;
+
+import java.util.List;
+
+@Service("betService")
+public class BetServiceImpl implements BetService {
+
+    private final UserRepository userRepository;
+    private final BetRepository betRepository;
+    private static final Logger logger = LoggerFactory.getLogger(BetServiceImpl.class);
+    private final UserLevelService userLevelService;
+
+    @Autowired
+    public BetServiceImpl(UserRepository userRepository, BetRepository betRepository,
+                          UserLevelService userLevelService) {
+        this.userRepository = userRepository;
+        this.betRepository = betRepository;
+        this.userLevelService = userLevelService;
+    }
+
+    @Override
+    public void withdrawBetMoneyAndExpForMatch(Match match) {
+        List<Bet> bets = betRepository.findAllByMatch(match);
+        bets.forEach(b -> {
+            User user = b.getUser();
+            int userStake = b.isSelectedA() ? match.getStakeA() : match.getStakeB();
+            boolean userWon = checkIfUserWon(b);
+            if (userWon) {
+                user.addCoins(b.getCoins() * userStake);
+            }
+            userLevelService.addExpAfterBet(user, userWon);
```

Comment:
Tutaj jeszcze bym proponował, żeby operacje dodania coinsów i expa wrzucić do jednej metody w serwisie.
Może nawet jakby się nad tym zastanowić, to można by z tego zrobić wzorzec obserwatora.

Reply:
Można by było, chociaż celowo rozdzieliłem exp i coiny xD

Reply:
Bardziej chodziło mi o to, żeby te dwie metody ```addCoins()``` i ```addExpAfterBet()``` wywoływać w jakiejś metodzie, która jest wywoływana bo skończeniu zakładu i tylko ta jedna metoda jest wywoływana.
Teraz jest tak dziwnie, że coinsy dodajemy do klasy ```User``` a exp do klasy ```userLevelService```.
Gdyby te dwie operacje wrzucić do jednej metody, to ukryjemy tę "dziwnaczność"